### DISCUSSION TITLE
drivers:ethernet:eth_w5500 fixes,command read_back/reset pin toggle #46139

### DISCUSSION THREAD

**Discussion by Hari-haranb** [zephyrproject-rtos/zephyr#46405] [RFC]
@parthitce @rlubos @lucasgcbkhomp  according to this PR  #46139 I tried to pulling low the reset pin after sleep time.i thought it doesn't make any issues & its works properly. but  unfortunately it causes RTR Read problem.so i consider the suggestions to reverting that particular reset pin toggle changes.

**Comment by Hari-haranb** [comment]
@lucasgcbkhomp any suggestions regarding this discussion ?

**Reply by lucasgcbkhomp** [reply]
@parthitce @Hari-haranb I participated in debugging the W5500 driver before the latest version which lead to @jancoow finding a solution with PR https://github.com/zephyrproject-rtos/zephyr/pull/45765 and netting us a more stable version of the driver. We had a few people testing it as you can see from the discussions from both the PR on top of the Issue thread, and it had been working fine since.

It then stopped working 100% during the initialization routine after PR https://github.com/zephyrproject-rtos/zephyr/pull/46139 according to my tests. Debugging the current driver it ocurred to me the GPIO logic is now inverted during initialization; from my understanding with the current version it should only successfully initialize if the reset pin on your board's `.dst` is set to `GPIO_ACTIVE_HIGH`. 

The implication of this change is that, according to the documentation, the reset pin is supposed to be `GPIO_ACTIVE_LOW`. This swap in the initialization will surely break any board application that has been using the driver.

I managed to fix it by swapping the `set` operations for the GPIO so the routine finishes with the reset pin in the low state as it used to:
```c
gpio_pin_set_dt(&config->reset, 1);
k_usleep(500);
gpio_pin_set_dt(&config->reset, 0);
```

That said, all this is exclusively about the potential initialization fault that was introduced. I do not know what the rest of the changes are meant to accomplish because I have no use case to test against and therefore I have no opinion. If it were up to me I'd revert the whole thing from lack of sufficient context.
